### ISSUE TITLE
Plan: merge recogbase into recog

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/recogbase#1]
I plan to merge recogbase into recog. There seems to be no real benefit to keep them separate, and it makes it difficult to perform badly needed cleanup and refactoring, as one has to coordinate two packages.

My plan is to retain as much of the history of `recogbase`, by first subtree merging it (with full history) into a subdirectory of `recog`, then, in a second commit `git mv` files around etc. until there is a single working package. I'll also add some subdirectories to the `gap` dir, to group things (e.g. the "base" code formerly in `recogbase`, vs. perm code, vs. matrix code.

**Comment by fingolfin** [comment]
@frankluebeck @ssiccha any comments or concerns?

**Comment by fingolfin** [comment]
Also @hulpke

**Comment by neunhoef** [comment]
I do not have anything against it. The original reason for separation was simply authorship. The original `recogbase` authors were Akos and I. In `recog` everybody should be an author who contributed code. But as things stand now, this authorship business is no longer relevant, so please go ahead in whatever way is convenient for those people driving the development now.

**Comment by hulpke** [comment]
Fine with me -- users would -- before or after -- simply load `recog`. Thanks for doing this.  Alexander

**Comment by ssiccha** [comment]
I'll talk to Alice about this on Monday.

But AFAIK nothing speaks against it. There just should be  a clear division in the code and the documentation between the framework and the methods used for matrix recognition, to state the obvious.

**Comment by fingolfin** [comment]
Since feedback was positive, I went ahead and did it in a pull request: https://github.com/gap-packages/recog/pull/25

It also contains a few additional improvements to the (merged) manual etc. Sadly virtually nothing w discussed 1.5 years ago in Aachen (see <https://github.com/gap-packages/recog/wiki/Notes-from-meeting-in-Aachen,-April-2016>) was tackled by anybody so far; my hope is that by merging the two packages, at last there is only one place one has to watch. I'll see if I can at least improve some of the documentation and core code (I already worked a bit on that earlier today). 

I plan to merge this soon, but won't do this so later at night, w/o having slept on it ;-)

Once it is merged, I'll archive the `recobase` repository.

**Comment by fingolfin** [comment]
Added another minor but useful improvement: the table of methods is now generated from the source codes, thus it is always up-to-date. This also helps to get a better idea about which ones still need to be documented.

**Comment by aniemeyer** [comment]
Since Max Neunhoeffer has given it the go ahead I have no objections.

**Comment by frankluebeck** [comment]
Merging these packages into one is fine with me.

**Comment by fingolfin** [comment]
Done
